index.js

```diff
@@ -34,6 +35,9 @@ async function main() {
     url.searchParams.append('variables[GITHUB_REF_TYPE]', process.env.GITHUB_REF_TYPE);
     url.searchParams.append('variables[GITHUB_REPO]', `${repo.repo}`);
     url.searchParams.append('variables[GITHUB_SHA]', sha);
+    if (schedule != null) {
+        url.searchParams.append('variables[GITHUB_SCHEDULE]', schedule);
+    }
```

Comment:
IMO `schedule` could never be `null`, judging based on https://github.com/actions/toolkit/blob/main/packages/core/src/core.ts#L173 (actually looking at the code it looks like it just ignored `required` flag and throws an exception)

Have you checked that it's really `null`?

Reply:
No, I haven't checked it. I will see what will happen when the option is missing and report back here.

Reply:
Seems that the `getBooleanParameter` indeed doesn't support optional parameters: https://github.com/NordSecurity/libtelio/actions/runs/5795350690/job/15706748246#step:2:9
I will get back with a workaround...

Reply:
I've fixed it by defining default in the `action.yml` as mentioned in the bug: https://github.com/actions/toolkit/issues/844

Reply:
That’s strange, because it’s even documented as option-able: https://github.com/actions/toolkit/tree/main/packages/core#inputsoutputs. :face_exhaling:

Reply:
> That’s strange, because it’s even documented as option-able: https://github.com/actions/toolkit/tree/main/packages/core#inputsoutputs. face_exhaling

They do mention this:
> If required set to be false, the input should have a default value in action.yml.